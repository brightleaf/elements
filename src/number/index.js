import { Number } from './number'

export { Number }

export default Number
