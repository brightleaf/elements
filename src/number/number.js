import React from 'react'
import classnames from 'classnames'
import { Base } from '../base'

export const Number = ({ className, children, ...props }) => {
  return (
    <Base className={classnames('number', className)} {...props}>
      {children}
    </Base>
  )
}

Number.defaultProps = {
  as: 'div',
}

export default Number
