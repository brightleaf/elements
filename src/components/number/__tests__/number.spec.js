import React from 'react'
import { render, cleanup } from '@testing-library/react'
import {
  toHaveClass,
  toHaveAttribute,
  toContainElement,
} from '@testing-library/jest-dom'
import { Number } from '../number'

expect.extend({ toHaveClass, toHaveAttribute, toContainElement })
describe('The Number Component', () => {
  afterEach(cleanup)
  it('should render a div with the class number', () => {
    const { container } = render(<Number>2</Number>)
    expect(container.firstChild).toMatchSnapshot()
    expect(container.firstChild).toHaveClass('number')
  })
  it('should render a div with the class number and the className passed in', () => {
    const { container } = render(<Number className="test" />)
    expect(container.firstChild).toMatchSnapshot()
    expect(container.firstChild).toHaveClass('number')
    expect(container.firstChild).toHaveClass('test')
  })
  it('should render with color and size modifiers', () => {
    const { container } = render(
      <Number hasBackgroundPrimary hasTextWeightBold hasTextWhite>
        23
      </Number>
    )
    expect(container).toMatchSnapshot()

    expect(container.firstChild).toHaveClass('number')
    expect(container.firstChild).toHaveClass('has-text-weight-bold')
    expect(container.firstChild).toHaveClass('has-text-white')
    expect(container.firstChild).toHaveClass('has-background-primary')
  })
})
