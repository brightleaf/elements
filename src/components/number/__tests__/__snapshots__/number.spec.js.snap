// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Number Component should render a div with the class number 1`] = `
<div
  class="number"
>
  2
</div>
`;

exports[`The Number Component should render a div with the class number and the className passed in 1`] = `
<div
  class="number test"
/>
`;

exports[`The Number Component should render with color and size modifiers 1`] = `
<div>
  <div
    class="number has-background-primary has-text-white has-text-weight-bold"
  >
    23
  </div>
</div>
`;
